moz-extensions/src/differential/customfield/DifferentialUpliftRequestCustomField.php

```diff
@@ -69,6 +71,96 @@ public function shouldAppearInApplicationTransactions() {
     return true;
   }
 
+  /* Set QE required flag on the relevant Bugzilla bug. */
+  public function setManualQERequiredFlag(int $bug) {
+    // Construct request for setting `qe-verify` flag, see
+    // https://bmo.readthedocs.io/en/latest/api/core/v1/bug.html#update-bug
+    $url = id(new PhutilURI(PhabricatorEnv::getEnvConfig('bugzilla.url')))
+        ->setPath('/rest/bug/'.$bug);
+    $api_key = PhabricatorEnv::getEnvConfig('bugzilla.automation_api_key');
+
+    // Encode here because `setData` below will fail due to nested arrays.
+    $data = phutil_json_encode(
+        array(
+            'flags' => array(
+                array(
+                    'name' => 'qe-verify',
+                    'status' => '?',
+                ),
+            ),
+        )
+    );
+
+    $future = id(new HTTPSFuture($url))
+        ->addHeader('Accept', 'application/json')
+        ->addHeader('User-Agent', 'Phabricator')
+        ->addHeader('X-Bugzilla-API-Key', $api_key)
+        ->setData($data)
+        ->setMethod('PUT')
+        ->setTimeout(PhabricatorEnv::getEnvConfig('bugzilla.timeout'));
+
+    try {
+      list($status, $body) = $future->resolve();
+      $status_code = (int) $status->getStatusCode();
+
+      $thebody = $body;
+      # Return an error string and invalidate transaction if Bugzilla can't be contacted.
+        $body = phutil_json_decode($body);
+      if (array_key_exists("error", $body) && $body["error"]) {
+        $message = $body["message"];
+        throw new Exception(
+            'Could not set `qe-verify` on Bugzilla: '.$data.'! Status code: '.$status_code.' body: '.$thebody.'\nPlease file a bug.'
+        );
+      }
```

Comment:
This is the piece that doesn't work in suite. Before this lands I'll have to clean up some of the stuff in the exception error message, but for now it prints the status code and the object that was sent to Bugzilla.